### PR TITLE
Startup script & Frontend maven

### PR DESCRIPTION
**Pull request by Pwarde**
Created a script to easily start a springboot server together with gulp.
Use same config to also build new JAR.

Replacement for bash script made before.

Note that the missed-models-additions branch is also in this branch. So that one should be merged before this one.

Please squash commits on merge (I cannot find the option here)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,5 +39,14 @@ gitlab-tools/itest_farm.out
...
+custom-yamls/
+
+node_modules
+
+<<<<<<< HEAD
+config.yaml
+
+node_modules
+=======
+>>>>>>> 347ada181e204068f04d6a62cac01c252d03376f
```
Comment (lvandernoll) [inline .gitignore]: :O

File: babel.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+          edge: "17",
+          firefox: "60",
+          chrome: "67",
+          safari: "11.1",
+        },
+      },
+    ],
+  ];
+  
+  module.exports = { presets };
```
Comment (lvandernoll) [inline babel.config.js]: newline

File: config_example.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "STYLING_DEFAULTLOCALE": "nl_NL",
+    "SERVER_PORT": "2345"
+  },
+  "project3": {
+    "STYLING_EXTENSIONRESOURCESLOCATION": "<project folder>/target/classes/extension-resources",
+    "STYLING_EXTENSIONUNCOMPILEDLOCATION": "<project folder>/src/main/resources/extension-resources",
+    "STYLING_DEFAULTLOCALE": "nl_NL",
+    "SERVER_PORT": "9889"
+  }
+}
```
Comment (lvandernoll) [inline config_example.json]: newline

File: .gitignore [hunk 4]
```diff
@@ -39,5 +39,9 @@ gitlab-tools/itest_farm.out
...
 
 # Gulp
+custom-yamls/
+node_modules
 /gulp/node_modules
-/gulp/config.json
\ No newline at end of file
+/gulp/*.config.json
+config.json
+custom-yamls
```
Comment (jashaj) [inline .gitignore]: Is there both a file and a folder called `custom-yamls`? I guess line 42 was enough (it's 42 so it's the answer to life, universe and everything anyway).

File: babel.config.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const presets = [
+    [
+        "@babel/env",
+        {
+            targets: {
```
Comment (jashaj) [inline babel.config.js]: These targets get outdated quickly. Shouldn't we use something more generic?

File: example-yamls/personal__idin__select_bank.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#!! THIS DOESNT WORK YET !!
```
Comment (jashaj) [inline example-yamls/personal__idin__select_bank.yaml]: What does it take to make it work?

File: gulpfile.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Onegini B.V.
```
Comment (jashaj) [inline gulpfile.js]: It's 2020 ;-)

File: gulpfile.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+const gulp = require('gulp');
+const watch = require('gulp-watch');
+const bs = require('browser-sync').create();
+const sass = require('gulp-sass');
+const babel = require('gulp-babel');
+const uglify = require('gulp-uglify');
+const cleanCSS = require('gulp-clean-css');
+
+const compileSass = () => gulp.src(`${process.env.STYLING_EXTENSIONUNCOMPILEDLOCATION}/scss/**/*.scss`)
```
Comment (jashaj) [inline gulpfile.js]: `${process.env.STYLING_EXTENSIONUNCOMPILEDLOCATION}` ia ewpwaws multiple times. You may want to extract this to a const or a function that returms the path.

File: pom.xml [hunk 9]
```diff
@@ -210,7 +247,15 @@
         <version>${maven-surefire-plugin.version}</version>
       </plugin>
 
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
+        <configuration>
+          <source>1.8</source>
```
Comment (jashaj) [inline pom.xml]: Our other projects are on Java 11 now. Can we make this one use 11 as well?

File: .gitignore [hunk 10]
```diff
@@ -39,5 +39,9 @@ gitlab-tools/itest_farm.out
 gitlab-tools/__pycache__/*
 
 # Gulp
+custom-yamls/
+node_modules
```
Comment (jashaj) [inline .gitignore]: Not very likely but this would also ignore a file called node_modules

File: .gitignore [hunk 11]
```diff
@@ -39,5 +39,9 @@ gitlab-tools/itest_farm.out
...
 /gulp/node_modules
```
Comment (jashaj) [inline .gitignore]: I don't see a gulp directory. If it's meant for custom gulp files, then the entire directory should be ignored.

File: README.md [hunk 12]
```diff
@@ -149,27 +149,33 @@ The `standalone-styling` will be available under the Spring Boot Dashboard.
...
-1. If you use Windows, use [git bash](https://gitforwindows.org/) to run the script.
-2. Create a config file with the correct locations of your files (see example.cfg)
-3. Start script by running:
-  ```
-  bash start-standalone-styling.sh <config file>
-  ```
-  Replace \<config file\> with the config file you have made for the project you want to open
-4. Stop the script with `ctrl-c`
\ No newline at end of file
+For front-end development the springboot server can be run together with the Gulp taskrunner to see changes immediately.
```
Comment (jashaj) [inline README.md]: Spring Boot

File: pom.xml (hunk 9 above)
Comment (Pwarde) [reply pom.xml]: That is not the same as the version number of this plugin is it? I am not really aware of Java versions or where to define them in this pom file.

Comment (jashaj) [reply pom.xml]: This is exactly where you configure the Java compiler to expect that the source is Java 11 and that it should compile the result to Java 11 byte code. I'll change it here.

File: example-yamls/personal__idin__select_bank.yaml (hunk 6 above)
Comment (jashaj) [reply example-yamls/personal__idin__select_bank.yaml]: @lvandernoll can you check whether this comment is still applicable? If so what does it take to make it work?

**Pull request merged**